stingray/crossspectrum.py

```diff
@@ -1690,13 +1769,23 @@ def __init__(
             self.n = None
             return
 
-        # The legacy interface is used when the user asks for it, or for
-        # input objects that are not EventList or Lightcurve (e.g. lists of
-        # light curves.)
-        accepted_new_style = (EventList, Lightcurve)
-        legacy = legacy or not isinstance(data1, accepted_new_style)
+        if isinstance(data1, Generator):
+            warnings.warn(
+                "The averaged Cross spectrum from a generator of "
+                "light curves pre-allocates the full list of light "
+                "curves, losing all advantage of lazy loading. If it "
+                "is important for you, use the "
+                "AveragedCrossspectrum.from_lc_iterable static "
+                "method, specifying the sampling time `dt`.")
```

Comment:
Why do you need to specify a sampling time when you already have some kind of iterable of light curves?

Reply:
Because you need the dt before loading the light curves, and if you use lazy loading this cannot be done. There is a way to do that, but I would have had to change the interface yet again